[IS] Add docs for dynamic password migration (#4870)
wso2/docs-is: en/includes/guides/users/migrate-users/migrate-passwords.md

# Migrate passwords to {{ product_name }}

You may use one of the following two ways to migrate user passwords to {{ product_name }}.

## Migrate with password reset

This method involves the least administrative effort but causes the most friction for users. With a password reset, users will receive an email prompting them to set a password for their new {{ product_name }} accounts. Until they set a password, their user accounts will remain locked to prevent unauthorized access. Additionally, this method allows you to immediately terminate operations on your previous IdP once the user accounts are successfully migrated to {{ product_name }}.

{% if product_name == "Asgardeo" %}

If you wish to migrate user passwords using this method, inform the Asgardeo team along with your request to [migrate users]({{base_path}}/guides/users/migrate-users/migrate-users/).

{% endif %}

## Migrate with on-demand silent migration

> Compared to password reset method above, on-demand silent password migration involves more effort to set up but provides the user with a seamless password migration experience. The following are the high level steps involved with this method.

Reviewer: ```suggestion
Compared to the password reset method above, on-demand silent password migration involves more effort to set up but provides the user with a seamless password migration experience. The following are the high level steps involved with this method.
```

1. The user attempts to log in to the application.

2. The application redirects the user to the {{ product_name }} login screen.

3. The user enters the usual credentials for the legacy IdP.

4. If {{ product_name }} has already migrated the user's password, the user is logged in. If not, {{ product_name }} makes an authentication request to the legacy IdP with the user provided credentials.

5. {{ product_name }} redirects the user to a waiting page until the legacy IdP completes the authentication.

> 6. Once complete, the user is taken back to the application and {{ product_name }} silently migrates the password for the user.

Reviewer: ```suggestion
6. Once the process is complete, the user is redirected back to the application and {{ product_name }} silently migrates the password for the user.
```

7. The user is successfully logged in and may now continue using the usual credentials to log in to the application with {{ product_name }}.

!!! note

    If you wish to migrate users with this method, it is crucial to note the following:

     - The legacy IdP should run alongside {{ product_name }} for a set period to facilitate user migration.
     - Have a contingency plan to migrate any accounts that were not migrated while the legacy IdP is active.
     - Unmigrated users must contact the {{ product_name }} organization administrators to initiate a password reset process, facilitating a smooth and complete transition to {{ product_name }}.

The following guides go into detail on implementing on-demand silent password migration.

> ### How it works?

Reviewer: ```suggestion
### How it works
```
